app/routes/route_utilities.py

```diff
@@ -0,0 +1,109 @@
+from flask import abort, make_response, Response
+from ..db import db
+
+
+def validate_model(cls, model_id):
+    try: 
+        model_id = int(model_id)
+    except:
+        response = {"message": f"{cls.__name__} id{model_id} is invalid" }
+        abort(make_response(response, 400))
+    
+    query = db.select(cls).where(cls.id == model_id)
+    model = db.session.scalar(query)
+
+    if not model:
+        response = {"message": f"{cls.__name__} {model_id} not found"}
+        abort(make_response(response, 404))
+    
+    return model
+
+
+def create_model(cls, model_data):
+    try:
+        new_model = cls.from_dict(model_data)
+    except KeyError as error:
+        response = {"details": "Invalid data"}
```

Comment:
It might be nice to provide even more details in the response you send back so that the client knows how to fix up the request body they will re-send.

```suggestion
        response = {"details": f"Invalid data for field: {error.args[0]}"}
```
For example, this would evaluate to "title" for example if I sent a bad request body to create a Task.